packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -1181,7 +1181,11 @@ actual object RealmInterop {
     }
 
     actual fun realm_sync_config_new(user: RealmUserPointer, partition: String): RealmSyncConfigurationPointer {
-        return LongPointerWrapper(realmc.realm_sync_config_new(user.cptr(), partition))
+        return LongPointerWrapper<RealmSyncConfigT>(realmc.realm_sync_config_new(user.cptr(), partition)).also { ptr ->
+            // Stop the session immediately when the Realm is closed, so the lifecycle of the
+            // Sync Client thread is manageable.
+            realmc.realm_sync_config_set_session_stop_policy(ptr.cptr(), realm_sync_session_stop_policy_e.RLM_SYNC_SESSION_STOP_POLICY_IMMEDIATELY)
```

Comment:
We have previously tried to put all configuration in the library to ensure that we have the exact same configuration on all platforms. I guess it is quite tedious with having to wrap the enums, so wouldn't block approval, but could consider to do a no-arg-method that set it to `IMMEDIATELY` ... but I guess that could also be implemented incorrectly on each platform 🤪  ... but as a reviewer it would be easier to spot errors in that, than spotting that you have forgotten the same configuration call on each platform.

Reply:
Yes, that is the default, but if you are offline it will keep running indefinitely preventing the Sync thread from exiting, and thus the JVM.